solutions/standard/provider.tf

```diff
@@ -0,0 +1,34 @@
+########################################################################################################################
+# Provider config
+########################################################################################################################
+
+provider "ibm" {
+  ibmcloud_api_key = var.ibmcloud_api_key
+  region           = "us-south"
+}
+
+provider "ibm" {
+  alias            = "namespace"
+  ibmcloud_api_key = var.ibmcloud_api_key
+  region           = var.namespace_region
+}
```

Comment:
The IBM provider supports passing a value for [visibility](https://registry.terraform.io/providers/IBM-Cloud/ibm/latest/docs#visibility). The default is "public" which means public endpoint is always used for api calls.
We should probably expose this value as a variable, and possibly default it to "private", especially since there may be CBR rules in place that only allow services to be talked to via the private network.

Reply:
added new variable in da

```
variable "provider_visibility" {
  description = "The visibility to IBM Cloud endpoint. Allowable values are public, private, public-and-private."
  type        = string
  default     = "private"

  validation {
    condition     = contains(["public", "private", "public-and-private"], var.provider_visibility)
    error_message = "Invalid visibility option. Allowed values are 'public', 'private', or 'public-and-private'."
  }
}
```